### PR TITLE
1501 Docker implementation of federated iRODS zones for HydroShare

### PR DESCRIPTION
**Pull request by mjstealey**
Implement federated iRODS servers in Docker space for HydroShare
- replicates hydroshareZone and hydroshareuserZone
- follows the same workflow as found in production
- enabled by running one additional script prior to using `hsctl`
- see [README.md](https://github.com/hydroshare/hydroshare/tree/1501-docker-irods/irods) for usage details

### CODE REVIEW

**Comment by pkdash** [comment]
@hyi Can you review this?

File: hsctl [hunk 1]
```diff
@@ -227,6 +231,8 @@ loaddb_hs() {
 
 migrate_hs() {
     echo "MIGRATE:"
+    echo "  - docker exec -u hydro-service hydroshare python manage.py migrate sites"
+    docker exec -u hydro-service hydroshare python manage.py migrate sites
```
Comment (hyi) [inline hsctl]: @mjstealey Just out of curiosity, what does this ```migrate sites``` do?

**Comment by hyi** [comment]
@mjstealey I have pulled this branch to my local environment and tested the whole workflow and everything works as expected. Nice work! +1 to merge. Once this work is merged into develop, I can start to work on adding unit tests for federation-related functionalities using this federated irods docker containers as the irods test environment.

File: hsctl (hunk 1 above)
Comment (mjstealey) [inline hsctl]: @hyi - had occasionally run into a site ID issue on builds using a copy of the www db. This seemed to resolve it though it shouldn't technically be necessary... More often than not it is not required.

**Pull request merged**